### PR TITLE
adding my keymap

### PR DESCRIPTION
**Pull request by defcxz**
Just added my keymap! :)

### CODE REVIEW

File: keyboards/lily58/keymaps/defcxz/config.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+#pragma once
+
+/* Select hand configuration */
+
+#define MASTER_LEFT
+// #define MASTER_RIGHT
+// #define EE_HANDS
+
+#define USE_SERIAL_PD2
```
Comment (waffle87) [inline keyboards/lily58/keymaps/defcxz/config.h:29]:
```suggestion
```

File: keyboards/lily58/keymaps/defcxz/config.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#define SPLIT_LAYER_STATE_ENABLE
+#define SPLIT_MODS_ENABLE     // para que los modificadores funcionen en ambos lados
+
+#define TAPPING_FORCE_HOLD    // para que no se repita la tecla al mantenerla
+#define TAPPING_TERM 180      // cantidad en ms para considerar tap dance
+#define TAPPING_TERM_PER_KEY  // para que cada tecla tenga su propio tapping term
+
+#define OLED_BRIGHTNESS 255
+#define OLED_TIMEOUT 0
+#define RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/lily58/keymaps/defcxz/config.h:40]:
```suggestion
#define OLED_TIMEOUT 0

#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
```

File: keyboards/lily58/keymaps/defcxz/keymap.c [hunk 3]
```diff
@@ -0,0 +1,243 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/lily58/keymaps/defcxz/keymap.c:1]: This file is missing a license header.

File: keyboards/lily58/keymaps/defcxz/keymap.c [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+  KC_LSFT, KC_PGDN, KC_PSCR, _______, RGB_SAI, RGB_SAD,                   KC_F13,  KC_F14,  KC_F15,  KC_F16,  KC_F17,  KC_SLSH,
+  RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, _______, _______, KC_F19,  KC_F20,  KC_F21,  KC_F22,  KC_F23,  KC_F24,
+                             _______, KC_CAPS, _______, _______, _______, KC_DEL, KC_PLUS, KC_EQL
+),
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
```
Comment (waffle87) [inline keyboards/lily58/keymaps/defcxz/keymap.c:75]: This core [tri layer](https://docs.qmk.fm/#/feature_tri_layer) should be used here instead.
```suggestion
```

File: keyboards/lily58/keymaps/defcxz/rules.mk [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+MAGIC_ENABLE = no           # Enable magic
+
+# If you want to change the display of OLED, you need to change here
+SRC +=  ./lib/rgb_state_reader.c \
+        ./lib/layer_state_reader.c \
+        ./lib/keylogger.c \
+        # ./lib/timelogger.c \
+        # ./lib/logo_reader.c \
+        # ./lib/mode_icon_reader.c \
+        # ./lib/host_led_state_reader.c \
```
Comment (waffle87) [inline keyboards/lily58/keymaps/defcxz/rules.mk:20]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
SWAP_HANDS_ENABLE = no      # Enable one-hand typing
OLED_ENABLE = yes            # OLED display
WPM_ENABLE = yes            # Enable WPM counter
TAP_DANCE_ENABLE = yes      # Enable Tap dance
COMMAND_ENABLE = no         # Enable command
UNICODE_ENABLE = no         # Enable unicode
CONSOLE_ENABLE = no         # Enable console
MAGIC_ENABLE = no           # Enable magic
LTO_ENABLE = yes

SRC +=  ./lib/rgb_state_reader.c \
        ./lib/layer_state_reader.c \
        ./lib/keylogger.c
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
In the future, please fill out the pull request template that was provided.

File: keyboards/lily58/keymaps/defcxz/keymap.c (hunk 4 above)
Comment (drashna) [reply keyboards/lily58/keymaps/defcxz/keymap.c:90]: Either may be fine, but as MO is just being used, the tri layer keycodes would probably be better, yeah

**Comment by zvecr** [comment]
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**